rasa_nlu_examples/classifiers/sparse_sklearn_intent_classifier.py

```diff
@@ -0,0 +1,275 @@
+from rasa.shared.nlu.training_data.message import Message
+from rasa.shared.nlu.constants import (
+    INTENT,
+    INTENT_NAME_KEY,
+    PREDICTED_CONFIDENCE_KEY,
+)
+
+logger = logging.getLogger(__name__)
+
+if typing.TYPE_CHECKING:
+    import sklearn
+
+
+class SparseSklearnIntentClassifier(IntentClassifier):
+    r"""Base class for intent classifiers using the sklearn framework with sparse features.
+    Note that all sparse features will be used, i.e. there is no filtering for specific
+    featurizers yet."""
+
+    defaults = {}
+
+    @classmethod
+    def required_packages(cls) -> List[Text]:
+        return ["sklearn"]
+
+    @classmethod
+    def required_components(cls) -> List[Type[Component]]:
+        return [SparseFeaturizer]
+
+    def __init__(
+        self,
+        component_config: Optional[Dict[Text, Any]] = None,
+        clf: Optional["sklearn.base.ClassifierMixin"] = None,
+        le: Optional["sklearn.preprocessing.LabelEncoder"] = None,
+    ) -> None:
+        """Construct a new naive Bayes intent classifier using the sklearn framework."""
+        super().__init__(component_config)
+
+        if (not clf) + (not le) == 1:
+            raise ValueError(
+                "Expected classifier and label encoder instance. "
+                "Specifying only one of them leads to an incomplete model definition."
+            )
+        self.le = le
+        self.clf = clf
+
+    @abstractmethod
+    def create_sklearn_classifier(
+        self, **kwargs: Any
+    ) -> "sklearn.base.ClassifierMixin":
+        r"""Lazily imports the required sklearn classifier class and creates and
+        instantiates the sklearn classifier using all the given keyword arguments.
+
+        :param **kwargs: see defaults dictionary
+        """
+        pass
+
+    def train(
+        self,
+        training_data: TrainingData,
+        config: Optional[RasaNLUModelConfig] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Trains the intent classifier from scratch on a data set. A any classifier
+        components passed to the constructor will be forgotten, i.e.
+        you cannot use it for finetuning yet.
+
+        Reminder: Enabling finetuning here would be possible for sklearn
+        models that implement partial_fit or have a warm_start option.
+
+        :param config: unused
+        :param **kwargs: unused
+        """
+        if self.clf is not None:
+            rasa.shared.utils.io.raise_warning(
+                "Intent classifier has already been trained. Calling train again discards the old training results."
+            )
+
+        self.clf = self.create_sklearn_classifier(
+            **{key: self.component_config[key] for key in self.defaults}
+        )
+
+        X, y, self.le = self._collect_features_and_encode_labels(training_data)
+
+        if len(y):  # i.e. prepared data is not empty
+            with warnings.catch_warnings():
+                # filter out all sklearn warnigns
```

Comment:
Why do we want to filter all of the warnings? Is it really that verbose?

Reply:
Chris had introduced this because of some specific "UndefinedMetricWarning" - for LR, I know that one can get warnings about non-convergence a lot. We could make it optional - i.e. introduce some "verbose" in the defaults?

Reply:
Ah clear! 

Adding a verbose method seems fine, but the base behaviour of catching warnings seems fine. At times sklearn can get overly exited when it comes to logging so I can't say I mind this default.